src/workers/parser/getSymbols.js

```diff
@@ -255,15 +255,27 @@ function extractSymbol(path: SimplePath, symbols) {
 
   if (t.isVariableDeclarator(path)) {
     const node = path.node.id;
-    const { start, end } = path.node.loc;
+    const { properties } = node;
     if (t.isArrayPattern(node)) {
       return;
     }
-    symbols.identifiers.push({
-      name: node.name,
-      expression: node.name,
-      location: { start, end }
-    });
+    if (properties != undefined && t.objectPattern(properties)) {
+      properties.forEach(function(property) {
+        const { start, end } = property.loc;
+        symbols.identifiers.push({
+          name: property.value.name,
+          expression: property.value.name,
+          location: { start, end }
+        });
+      });
+    } else {
+      const { start, end } = path.node.loc;
+      symbols.identifiers.push({
+        name: node.name,
+        expression: node.name,
+        location: { start, end }
+      });
+    }
```

Comment:
it'd be nice to consolidate this and do something like:

```js
const properties = t.isObjectPattern(node.properties) ? node.properties: [path.node];
properties.forEach((property) => {
  const { start, end } = property.loc;
  // ...
})
```

Reply:
A draft version of the changes:
````
  if (t.isVariableDeclarator(path)) {
    const node = path.node.id;
    const { properties } = node;
    const hasProperties = (properties && t.objectPattern(properties));
    if (t.isArrayPattern(node)) {
      return;
    }
    const listProperties = hasProperties ? node.properties : [path.node];
      listProperties.forEach(function(property) {
      const { start, end } = property.loc;
      symbols.identifiers.push({
        name: hasProperties ? property.value.name : property.name,
        expression: hasProperties ? property.value.name : property.name,
        location: { start, end }
      });
    });
...
````
But these changes 'amplify' the `TypeError: undefined has no properties` issue:

With changes | Without changes
------------ | -------------
![b177f9ea-ce85-4c3d-b80f-b3d7d2ed141a](https://user-images.githubusercontent.com/36955296/38358492-a520bcf4-38c5-11e8-91ca-0a52e0ba8299.jpg) | ![96137cca-a833-4901-b324-10da91027124](https://user-images.githubusercontent.com/36955296/38358500-b03fe88a-38c5-11e8-8cd3-7831efaefae3.jpg)

This issue doesn't appear when I'm hovering `collection` in the same line.

Reply:
@marcogallo566 I would recommend you do a rebase first, there are some changes currently on master  that will fix the `undefined has no properties` issue and other things.

Reply:
I did it, but probably I did it wrong. I'll try again this evening, thank you!